3.10.0 does not compile

3.10.0 does not compile because:

```
    ./configure \
     --prefix=/usr \
     --sbindir=/usr/bin \
     --libexecdir=/usr/lib/cyrus \
     --sysconfdir=/etc \
     --enable-idled \
     --enable-nntp \
     --enable-backup \
     --disable-murder \
     --disable-http \
     --disable-calalarmd \
     --disable-replication \
     --disable-pcre \
     --disable-srs \
     --with-cyrus-user=cyrus \
     --with-syslogfacility=MAIL \
     --without-libwrap \
     --without-clamav

sieve/interp.c: In function 'extension_isactive':
sieve/interp.c:752:5: error: duplicate case value
  752 |     case SIEVE_CAPA_JMAPQUERY:
      |     ^~~~
sieve/interp.c:671:5: note: previously used here
  671 |     case SIEVE_CAPA_NOTIFY:
      |     ^~~~
make[2]: *** [Makefile:6771: sieve/libcyrus_sieve_la-interp.lo] Error 1
make[2]: Leaving directory '/build/cyrus-imapd/src/cyrus-imapd-3.10.0'
make[1]: *** [Makefile:7673: all-recursive] Error 1
make[1]: Leaving directory '/build/cyrus-imapd/src/cyrus-imapd-3.10.0'
make: *** [Makefile:3412: all] Error 2
```

Am 19.08.24 um 17:37 schrieb sandu1teodor:
> @pieska <https://github.com/pieska>, dose it work for you now? i get this when i try to compile 'make: *** No targets specified and no makefile found. Stop.
> '
works for me, compiles with commit 8442b53e7f1a592da7d2dcd26fd3670f908e90ae

-ap

--- a/sieve/interp.c
+++ b/sieve/interp.c
@@ -615,9 +615,11 @@ unsigned long long extension_isactive(sieve_interp_t *interp, const char *str)
               (config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_FILEINTO))) capa = 0;
         break;
 
+#ifdef ENABLE_REGEX
     case SIEVE_CAPA_REGEX:
         if (!(config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_REGEX)) capa = 0;
         break;
+#endif
 
     case SIEVE_CAPA_COPY:
         if (!(config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_COPY)) capa = 0;
@@ -674,7 +676,6 @@ unsigned long long extension_isactive(sieve_interp_t *interp, const char *str)
         break;
 
     case SIEVE_CAPA_ENOTIFY:
-    case SIEVE_CAPA_NOTIFY:
         if (!(interp->notify &&
               (config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_NOTIFY))) capa = 0;
         break;
@@ -755,11 +756,13 @@ unsigned long long extension_isactive(sieve_interp_t *interp, const char *str)
               (config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_VND_CYRUS_LOG))) capa = 0;
         break;
 
+#ifdef WITH_JMAP
     case SIEVE_CAPA_JMAPQUERY:
         if (!(interp->jmapquery &&
               (config_ext & IMAP_ENUM_SIEVE_EXTENSIONS_VND_CYRUS_JMAPQUERY)))
             capa = 0;
         break;
+#endif
 
     case SIEVE_CAPA_SNOOZE:
         if (!(interp->snooze &&

